plugins/kubernetes-backend/src/service/KubernetesFetcher.ts

```diff
@@ -105,61 +105,68 @@ export class KubernetesClientBasedFetcher implements KubernetesFetcher {
     this.logger = logger;
   }
 
-  fetchObjectsByServiceId(
+  fetchObjectsForService(
     serviceId: string,
     clusterDetails: ClusterDetails,
     objectTypesToFetch: Set<KubernetesObjectTypes>,
+    labelSelector: string,
   ): Promise<FetchResponseWrapper> {
     const fetchResults = Array.from(objectTypesToFetch).map(type => {
-      return this.fetchByObjectType(serviceId, clusterDetails, type).catch(
-        captureKubernetesErrorsRethrowOthers,
-      );
+      return this.fetchByObjectType(
+        clusterDetails,
+        type,
+        labelSelector.length !== 0
+          ? labelSelector
+          : `backstage.io/kubernetes-id=${serviceId}`,
```

Comment:
can be condensed to `labelSelector || 'backstage.io/...'` because the empty string is falsy